src/db_process.c

```diff
@@ -277,30 +278,31 @@ static int convert_str_to_int(char *str, char *err_msg)
  */
 static int vm_info_query_callback(void *all_vm_info_ptr, int argc, char **argv, char **azColName)
 {
-        /* Database schema for vm has 12 columns */
-        assert(argc == 12);
+        /* Database schema for vm has 11 columns */
+        assert(argc == 11);
  
         vm_info vm_info_captured;
         vm_info_captured.start_addr = strdup(argv[0]);
         vm_info_captured.end_addr = strdup(argv[1]);
         vm_info_captured.mmap_path = strdup(argv[2]);
+	vm_info_captured.compart_id = convert_str_to_int(argv[3], "compart_id is invalid");
 
 	// Checking that the received values are valid integers
 	// as expected from the vm_info_captured data structure. 
 	// If the check fails, it means the data is invalid. Without
 	// a more sophisticated logic in place to handle invalid data, 
 	// we will just quit the program and let users to examine 
 	// and fix the data before running this function again.
-	vm_info_captured.kve_protection = convert_str_to_int(argv[3], "kve_protection is invalid");
-     	vm_info_captured.mmap_flags = convert_str_to_int(argv[4], "mmap_flags is invalid"); 
-        vm_info_captured.vnode_type = convert_str_to_int(argv[5], "vnode_type is invalid");
-
-	vm_info_captured.bss_addr = argv[6] == NULL ? NULL : strdup(argv[6]);
-        vm_info_captured.bss_size = argv[7] == NULL ? NULL : strdup(argv[7]);
-        vm_info_captured.plt_addr = argv[8] == NULL ? NULL : strdup(argv[8]);
-        vm_info_captured.plt_size = argv[9] == NULL ? NULL : strdup(argv[9]);
-        vm_info_captured.got_addr = argv[10] == NULL ? NULL : strdup(argv[10]);
-        vm_info_captured.got_size = argv[11] == NULL ? NULL : strdup(argv[11]);
+	vm_info_captured.kve_protection = convert_str_to_int(argv[4], "kve_protection is invalid");
+     	vm_info_captured.mmap_flags = convert_str_to_int(argv[5], "mmap_flags is invalid"); 
+        vm_info_captured.vnode_type = convert_str_to_int(argv[6], "vnode_type is invalid");
+
+	//vm_info_captured.bss_addr = argv[7] == NULL ? NULL : strdup(argv[7]);
+        //vm_info_captured.bss_size = argv[8] == NULL ? NULL : strdup(argv[8]);
+        vm_info_captured.plt_addr = argv[7] == NULL ? NULL : strdup(argv[7]);
```

Comment:
I'd make `argv[7] == NULL ? NULL : strdup(argv[7]);` `_strdupnull(argv[7])` where `_strdupnull` is a local utility function. I'd then replace all the numbers with a `int i=0` at the top of the function and `i++` at each use. Then changing your mind about table structure becomes easier.